internal/controller/pipelinerollout/pipelinerollout_recycle.go

```diff
@@ -734,3 +704,70 @@ func isPipelineSpecOriginal(pipeline *unstructured.Unstructured) bool {
 	forceDrainSpecs, found := pipeline.GetAnnotations()[common.AnnotationKeyForceDrainSpecsStarted]
 	return !found || forceDrainSpecs == ""
 }
+
+// shouldDeleteRecyclablePipeline determines if a recyclable pipeline should be deleted immediately
+// without going through the drain process. Returns true if the pipeline should be deleted.
+func (r *PipelineRolloutReconciler) shouldDeleteRecyclablePipeline(
+	ctx context.Context,
+	pipelineRollout *apiv1.PipelineRollout,
+	pipeline *unstructured.Unstructured) (bool, error) {
+	numaLogger := logger.FromContext(ctx)
+
+	if pipelineRollout.Spec.Strategy != nil && pipelineRollout.Spec.Strategy.Progressive.ForcePromote {
+		// this is a case in which user likely doesn't care about data loss (and also we have no idea if the "promoted" pipeline is any good), so we can just delete the pipeline without draining
+		numaLogger.Debug("PipelineRollout strategy implies no concern for data loss so Pipeline will be deleted without draining")
+		r.registerFinalDrainStatus(pipelineRollout.Namespace, pipelineRollout.Name, pipeline, false, metrics.LabelValueDrainResult_DrainNotRequired)
+		return true, nil
+	}
+
+	// if pipeline doesn't require drain then just delete it
+	if pipeline.GetAnnotations() == nil || pipeline.GetAnnotations()[common.AnnotationKeyRequiresDrain] != "true" {
+		numaLogger.Debug("Pipeline does not require drain and will be deleted now")
+		r.registerFinalDrainStatus(pipelineRollout.Namespace, pipelineRollout.Name, pipeline, false, metrics.LabelValueDrainResult_DrainNotRequired)
+		return true, nil
+	}
+
+	expired, err := r.checkForRecycleExpirationTime(ctx, pipeline)
+	if err != nil {
+		return false, err
+	}
+	if expired {
+		numaLogger.Info("Pipeline has reached the max recyclable duration and will be deleted now")
+		r.registerFinalDrainStatus(pipelineRollout.Namespace, pipelineRollout.Name, pipeline, false, metrics.LabelValueDrainResult_NeverDrained)
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// checkForRecycleExpirationTime checks if the max recyclable duration has been exceeded.
+// Returns true if the pipeline should be deleted due to expiration.
+func (r *PipelineRolloutReconciler) checkForRecycleExpirationTime(
```

Comment:
`isRecycleExpired()` fn name can be more meaningful?

Reply:
cool, I like the conciseness of that, will change